src/App.tsx

```diff
@@ -1,14 +1,84 @@
 /* eslint-disable max-len */
-import React from 'react';
+// eslint-disable-next-line object-curly-newline
+import React, { ChangeEvent, useEffect, useMemo, useState } from 'react';
+
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { SortType } from './types/SortType';
+
+const getVisibleTodos = (filterCase: SortType, todos: Todo[], searchQuery: string) => {
+  let searchedTodo;
+
+  if (searchQuery.trim()) {
+    searchedTodo = todos.filter(todo => todo.title.toLowerCase().includes(searchQuery.trim().toLowerCase()));
+  }
+
+  switch (filterCase) {
+    case SortType.ACTIVE:
+      return searchedTodo === undefined
+        ? todos.filter(todo => !todo.completed)
+        : searchedTodo.filter(todo => !todo.completed);
+
+    case SortType.COMPLETED:
+      return searchedTodo === undefined
+        ? todos.filter(todo => todo.completed)
+        : searchedTodo.filter(todo => todo.completed);
+
+    default:
+      return searchedTodo === undefined ? todos : searchedTodo;
+  }
+};
 
 export const App: React.FC = () => {
+  const [allTodos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState(SortType.ALL);
+  const [query, setQuery] = useState('');
+  const [modalInfo, setModalInfo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    getTodos().then(todos => {
+      setTodos(todos);
+      setIsLoading(true);
+    });
+    setIsLoading(false);
+  }, []);
+
+  const handleChangeModalInfo = (todo: Todo | null) => {
+    setModalInfo(todo);
+  };
+
+  const filterChange = (event: ChangeEvent<HTMLSelectElement>) => {
+    if (event.target.value === 'all') {
```

Comment:
better to use switch 

and use enum values for comparison instead of strings `all, active, completed`